invisible_cities/cities/detsim_simulate_electrons.py

```diff
@@ -0,0 +1,52 @@
+import numpy  as np
+
+from typing    import   List, Tuple, Callable
+
+#######################################
+######### ELECTRON SIMULATION #########
+#######################################
+def generate_electrons(energies    : np.array,
+                       wi          : float,
+                       fano_factor : float ) -> np.array:
+    """ generate secondary electrons from energy deposits
+    """
+    nes  = np.array(energies/wi, dtype = int)
+    pois = nes < 10
+    nes[ pois] = np.random.poisson(nes[pois])
+    nes[~pois] = np.round(np.random.normal(nes[~pois], np.sqrt(nes[~pois] * fano_factor)))
```

Comment:
I've never been fully convinced of the necessity of this differentiation. Maybe @paolafer or @gonzaponte have more insight. In any case, I think the `< 10` needs to be justified to some extent. Do you have a notebook or similar with plots? What difference do we get in distribution if we always use the fano factor? What change in performance?

Reply:
I think this comes from C++. Numpy handles different values of the mean of the distribution automatically. I haven't checked the performance, though.

P.S: The value chosen by numpy is also 10.

Reply:
You mean that if you use a mean value greater than 10 with `np.random.poisson` it really uses `np.random.normal`? So in our case the only real difference if we just used normal is that the standard deviation will be scaled by a factor of `sqrt(fano_factor)` for signals below 10 pe?

Reply:
No. For values of the mean greater than 10 it uses a different algorithm, but it would still be a Poisson distribution.

But I had forgotten about the fano factor.  I was thinking this in the wrong way (assuming we used Gauss just as a limit of the Poisson for large values of the mean). We actually want to generate a normal distribution with a smaller standard deviation than the one coming from the Poisson limit. Therefore we *only* generate Poisson when we cannot rely on the gaussian distribution. The problem with the Poisson distribution is that the std deviation and the mean are not independent and we cannot make it narrower as we do with the gaussian.

So in conclusion, hits with less than 10 ie will have a wider distribution than they should.

Reply:
Can't we just simply add the `fano_factor` in the poisson?

Reply:
No, that was precisely my point when I said _"with the Poisson distribution [...] the std deviation and the mean are not independent"_

Reply:
I don't know what's the statistical limit of the Fano factor. Does it work the same way for a population of just a few electrons? I wouldn't be surprised if it's never been measured for xenon gas at energies below 1 keV. 
In any case, if the product _NF_ is < 1 (which is the case when _N_<10), the resulting Poisson distribution yields most of the time either 0 or 1 electrons. The idea was to avoid that behavior, given that the _W_ energy is just an effective average.

Reply:
OK, so I think we probably all agree to leave it as is? But I'd add a comment with a bit of explanation of the choice, otherwise it's a bit confusing.

Reply:
Based on @jmalbos comment, the new implementation decides whether to "poissonize" or "normalizate" based on the variance NF<1. I think this criteria makes the function less confusing than the 10 electrons "dont't know where it comes" value.